### PR TITLE
BaSP-TG-34: Employee, added shared components

### PR DESCRIPTION
**Pull request by FabianDutraSilva**
- Add shared components (Button, Modal, Table, Row)
- Modify form so edit functionality work correctly
- Add routes to Layout (Edit and Delete functionalities)

image

### CODE REVIEW

File: src/Components/Shared/Row/index.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          );
+        } else {
+          return <td key={index}>{data[header]}</td>;
+        }
+      })}
+    </tr>
+  );
+};
+
+export default Row;
```
Comment (Juanoauil) [inline src/Components/Shared/Row/index.jsx:34]: this component was not already merge?

File: src/Components/Employees/Form/index.js [hunk 2]
```diff
@@ -75,8 +77,8 @@ const AddEmployee = () => {
         alert('Error, employee was not created', data);
       } else {
         alert('Employee added', data.message);
```
Comment (Juanoauil) [inline src/Components/Employees/Form/index.js:79]: are these alerts correct? you shouldnt implement shared modal?

File: src/Components/Employees/Form/index.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
 import React, { useState, useEffect } from 'react';
 import style from '../employees.module.css';
```
Comment (Juanoauil) [inline src/Components/Employees/Form/index.js:2]: please add styles to form

File: src/Components/Shared/Table/index.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          })
+        ) : (
+          <p className={styles.noData}>There is no data available</p>
+        )}
+      </tbody>
+    </table>
+  );
+};
+
+export default Table;
```
Comment (Juanoauil) [inline src/Components/Shared/Table/index.jsx:36]: this component was not already merge?

**Review by Juanoauil — CHANGES_REQUESTED** [review]

**Comment by Juanoauil** [comment]
please resolve conflict when you create a pr

File: src/Components/Employees/Form/index.js (hunk 2 above)
Comment (FabianDutraSilva) [reply src/Components/Employees/Form/index.js:79]: fixed

File: src/Components/Shared/Row/index.jsx (hunk 1 above)
Comment (FabianDutraSilva) [reply src/Components/Shared/Row/index.jsx:34]: fixed

File: src/Components/Shared/Table/index.jsx (hunk 4 above)
Comment (FabianDutraSilva) [reply src/Components/Shared/Table/index.jsx:36]: solved conflicts on this branch

File: src/Components/Employees/Form/index.js (hunk 3 above)
Comment (FabianDutraSilva) [reply src/Components/Employees/Form/index.js:2]: fixed

**Review by PilarFernandezC — APPROVED** [review]

**Review by Juanoauil — APPROVED** [review]

**Comment by SteampunkMatu** [comment]
Looks great but I need you to move the button add only below of the "Employee" title thus we maintain consistency with the rest of the entities. Do that when you have time

**Comment by mara666** [comment]
Okey! I fix it. Ty @SteampunkMatu

**Review by DamianPalavecino — APPROVED** [review]
Changes were made correctly, approved

**Pull request merged**